mslib/msui/mscolab.py

```diff
@@ -1928,6 +1928,13 @@ def logout(self):
             return
         self.ui.local_active = True
         self.ui.menu_handler()
+
+        # disconnect socket
+        if self.conn is not None:
+            self.conn.disconnect()
+            self.conn = None
+        QtWidgets.QApplication.processEvents()
```

Comment:
Please add a comment before the processEvents() line
e.g.

ToDo review why the code gets blocked and refactor this part


The problem with this is that we not know where it hangs and this cause needs to be resolved.
In the "released users code" we should never need QtWidgets.QApplication.processEvents()


good article about qthreadpool, I don't see from our site a thread involved. The article describes the behaviour.
https://www.pythonguis.com/tutorials/multithreading-pyqt-applications-qthreadpool/